### ISSUE TITLE
PHP 7 Compatibility based on PHP Compatibility Checker

### ISSUE THREAD

**Issue by phauer** [aramk/crayon-syntax-highlighter#378]
Hi everyone,

I investigated whether my wordpress installation is ready for PHP7. That's why I used the Plugin [PHP Compatibility Checker](https://de.wordpress.org/plugins/php-compatibility-checker/) which checks my plugins for PHP7-compatibility.

It says that Crayon Syntax Highlighter is not compatible due to the following error:
```
FILE: <path>/wp-content/plugins/crayon-syntax-highlighter/crayon_settings.class.php
---------------------------------------------------------------------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
---------------------------------------------------------------------------------------------------------------------------------
 87 | ERROR | Function name, class name, namespace name or constant name can not be reserved keyword 'MIXED' (since version 7.0)
---------------------------------------------------------------------------------------------------------------------------------
```
However, in my local php 7 test setup, the crayon-syntax-highlighter works just fine. It seems that this is not a critical problem.

I'm not a PHP expert, about it just seems to be a naming issue. Will this be fixed? Is this a blocker for my php7 migration? I suppose it's not.

**Comment by majorsl** [comment]
I'm seeing this as well.

**Comment by aramk** [comment]
There was a fix in the PR #381. Thanks for posting the error message. I'll merge it now and release shortly.

**Comment by aramk** [comment]
Please test 6443abecd845db75cb0cb8e7df5d590a55babba0

**Comment by dschense** [comment]
after patching the files i get:

[error] 15195#15195: *4 FastCGI sent in stderr: "PHP message: PHP Fatal error:  Uncaught Error: Undefined class constant 'ALTERNATE' in /crayon-syntax-highlighter/crayon_highlighter.class.php:161

**Comment by edwarddekker** [comment]
At my website do i Have no problems. It Sounds like a problem with your FastCGI engine. I do no Have Anny clue.

Maybe Will it helps when you showing your error-logs.

**Comment by dschense** [comment]
php give this error: 
[pool mypool] child 21892 exited on signal 6 (SIGABRT) after 33.059909 seconds from start

its the Plugin that crashes the site. after disabling the plugin everything is running. And its just the first site of the wordpress setup.

**Comment by edwarddekker** [comment]
You should download the latest master copy from this github website. Than are there no problems anymore. Because you are using a Version before 2.8.5.

**Comment by dschense** [comment]
its still not working.

this is my php 7 pool.d config maybe this is the problem:

[wordpress]
user = www-data
group = www-data
listen = /var/run/php/php7.0-fpm-wordpress.sock
listen.owner = www-data
listen.group = www-data
php_admin_value[disable_functions] = exec,passthru,shell_exec,system
php_admin_flag[allow_url_fopen] = off
pm = dynamic
pm.max_children = 5
pm.start_servers = 2
pm.min_spare_servers = 1
pm.max_spare_servers = 3
chdir = /

**Comment by edwarddekker** [comment]
I see that you runnig with fastCGI.
Could you also try it without FastCGI.
I hope that you'll see no errors.

At my configuration with Apache, PHP 7.0.3 and Perl 5.8.1 & Python 2.2.3 do I find no errors at all.

**Comment by aramk** [comment]
Yeah, please try the latest on the master branch and let me know if it resolved your issues. There shouldn't be an undefined constant `ALTERNATE` if it's set in `crayon_settings.class.php`.

**Comment by dschense** [comment]
@aramk 

still getting these errors after aktivating the Plugin.

```
[17-Oct-2016 11:38:00] WARNING: [pool www] child 3466 exited on signal 6 (SIGABRT) after 4.567786 seconds from start
[17-Oct-2016 11:38:00] NOTICE: [pool www] child 3488 started
```

its the latest master.zip

**Comment by aramk** [comment]
Are you able to find a backtrace of the issue within the PHP logs?
